ignite/metrics/entropy.py

```diff
@@ -0,0 +1,90 @@
+from typing import Sequence
+
+import torch
+import torch.nn.functional as F
+
+from ignite.exceptions import NotComputableError
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+__all__ = ["Entropy"]
+
+
+class Entropy(Metric):
+    r"""Calculates the mean of `entropy <https://en.wikipedia.org/wiki/Entropy_(information_theory)>`_.
+
+    .. math:: H = \frac{1}{N} \sum_{i=1}^N \sum_{c=1}^C -p_{i,c} \log p_{i,c},
+       \quad p_{i,c} = \frac{\exp(z_{i,c})}{\sum_{c'=1}^C \exp(z_{i,c'})}
+
+    where :math:`p_{i,c}` is the prediction probability of :math:`i`-th data belonging to the class :math:`c`.
+
+    - ``update`` must receive output of the form ``(y_pred, y)`` while ``y`` is not used in this metric.
+    - ``y_pred`` is expected to be the unnormalized logits for each class. :math:`(B, C)` (classification) or :math:`(B, C, ...)` (e.g., image segmentation) shapes are allowed.
```

Comment:
Code formatting tests are not passing: https://github.com/pytorch/ignite/actions/runs/8375820095/job/22974497359?pr=3210#step:11:89 due to this long line. 

Maybe we can split it like this:
```suggestion
    - ``y_pred`` is expected to be the unnormalized logits for each class. :math:`(B, C)` (classification) 
      or :math:`(B, C, ...)` (e.g., image segmentation) shapes are allowed.
```
two spaces on the next line after above `-`.

Reply:
Thank you for your correction!